includes/admin/tools/class-llms-admin-tool-limited-billing-order-locator.php

```diff
@@ -0,0 +1,285 @@
+				'Edit Link',
+			)
+		);
+
+		// Create the CSV file.
+		ob_start();
+		$fh = fopen( 'php://output', 'w' );
+		foreach ( $csv as $line ) {
+			fputcsv( $fh, $line );
+		}
+		fclose( $fh ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fclose
+
+		return ob_get_clean();
+
+	}
+
+	/**
+	 * Retrieve a description of the tool
+	 *
+	 * This is displayed on the right side of the tool's list before the button.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	protected function get_description() {
+
+		$count = count( $this->get_csv() );
+
+		$desc = sprintf(
+			// Translators: %1$s = opening anchor link to documentation; %2$s = closing anchor link.
+			__( 'The method used to determine when a limited-billing recurring order has completed its payment plan changed during version 5.3.0. This tool provides a report of orders which may been affected by this change. %1$sRead more%2$s about this change.', 'lifterlms' ),
+			'<a href="https://lifterlms.com/docs/###" target="_blank">',
+			'</a>'
+		);
+		$desc .= ' ';
+		// Translators: %d = the number of pending batches.
+		$desc .= sprintf(
+			_n(
+				'There is %d order that should be reviewed.',
+				'There are %d orders that should be reviewed.',
+				$count,
+				'lifterlms'
+			),
+			$count
+		);
+
+		return $desc;
+
+	}
+
+	/**
+	 * Retrieve the tool's label
+	 *
+	 * The label is the tool's title. It's displayed in the left column on the tool's list.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	protected function get_label() {
+		return __( 'Limited Billing Orders', 'lifterlms' );
+	}
+
+	/**
+	 * Retrieves an array representing a line in generated CSV for the given order
+	 *
+	 * An order is considered to be affected by the change if either of the following conditions are true:
+	 *
+	 *   + Any number of refunds exist for the order. Since we are now counting refunds towards the billing limit
+	 *     an active order with any number of refunds should be reviewed by the admin.
+	 *
+	 *   + The plan is marked as "ended" and the total number of successful payments is not equal to the billing length.
+	 *     In this scenario admins will likely want to add additional payments and start the order again.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Order $order The order object.
+	 * @return array Array of information on the order to be stored in the generated CSV or an empty array of the order
+	 *               was not affected by the change.
+	 */
+	protected function get_order_csv( $order ) {
+
+		$refunds   = $this->get_txn_count_by_status( $order, 'llms-txn-refunded' );
+		$successes = $this->get_txn_count_by_status( $order, 'llms-txn-succeeded' );
+		$total     = $refunds + $successes;
+
+		$ended    = llms_parse_bool( $order->get( 'plan_ended' ) );
+		$expected = $order->get( 'billing_length' );
+
+		if ( $refunds >= 1 || ( $ended && $total !== $expected ) ) {
+
+			$id   = $order->get( 'id' );
+			$link = get_edit_post_link( $id );
```

Comment:
```suggestion
			$link = get_edit_post_link( $id, 'raw' );
```
or anything different from 'display', which is the default value, which will turn & into &amp; generating something like:
https://mywp.loc/wp-admin/post.php?post=1389&amp;action=edit
which is a "bad" link.
That said, the WordPress doc says:
Optional. How to output the '&' character. Default '&amp;'.
https://core.trac.wordpress.org/browser/tags/5.8/src/wp-includes/link-template.php#L1433
but the generated html doc:
(string) (Optional) How to output the '&' character. Default '&'.
so I guess it's not your fault :P
cc @pondermatic maybe something you want to "consider" for your work on the docblocks?

Reply:
Yes, you're correct about this. Nice.

I also noticed the unit tests weren't actually testing anything with relation to this due to having no logged in user. Whoop. Fixed that too!